android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/ImageViewFactory.java

```diff
@@ -0,0 +1,86 @@
+package com.vijay.jsonwizard.widgets;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Color;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.vijay.jsonwizard.R;
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+import com.vijay.jsonwizard.fragments.JsonFormFragment;
+import com.vijay.jsonwizard.interfaces.CommonListener;
+import com.vijay.jsonwizard.interfaces.FormWidgetFactory;
+import com.vijay.jsonwizard.utils.FormUtils;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ImageViewFactory implements FormWidgetFactory {
+
+    private View rootLayout;
+
+    @Override
+    public List<View> getViewsFromJson(String stepName, Context context, JsonFormFragment formFragment, JSONObject jsonObject, CommonListener listener, boolean popup) throws Exception {
+        List<View> views = new ArrayList<>(1);
+        rootLayout = LayoutInflater.from(context).inflate(getLayout(), null);
+        setWidgetTags(jsonObject, context);
+        views.add(rootLayout);
+        return views;
+    }
+
+    @Override
+    public List<View> getViewsFromJson(String stepName, Context context, JsonFormFragment formFragment, JSONObject jsonObject, CommonListener listener) throws Exception {
+        return getViewsFromJson(stepName, context, formFragment, jsonObject, listener, false);
+    }
+
+    private int getLayout() {
+        return R.layout.native_form_image_view;
+    }
+
+    private void setWidgetTags(JSONObject jsonObject, Context context) throws JSONException {
+        String key = jsonObject.getString(JsonFormConstants.KEY);
+        String openMrsEntityParent = jsonObject.getString(JsonFormConstants.OPENMRS_ENTITY_PARENT);
```

Comment:
Replace jsonObject.optString(JsonFormConstants.OPENMRS_ENTITY_ID, ""); instead for all the `jsonObject.getString`